packages/aws-cdk/lib/api/accelerated-deployments.ts

```diff
@@ -0,0 +1,242 @@
+  // apply the short-circuitable changes
+  await applyAllAcceleratedChanges(sdk, stackArtifact, acceleratableChanges);
+
+  return { noOp: acceleratableChanges.length === 0, stackArn: cloudFormationStack.stackId, outputs: cloudFormationStack.outputs, stackArtifact };
+}
+
+function findAllAcceleratableChanges(
+  stackChanges: cfn_diff.TemplateDiff, assetParamsWithEnv: { [key: string]: string },
+): LambdaFunctionResource[] | undefined {
+  const acceleratableResources = new Array<LambdaFunctionResource>();
+  let foundNonAcceleratableChange = false;
+  stackChanges.resources.forEachDifference((logicalId: string, change: cfn_diff.ResourceDifference) => {
+    const lambdaFunctionShortCircuitChange = isAcceleratedLambdaFunctionChange(logicalId, change, assetParamsWithEnv);
+    if (lambdaFunctionShortCircuitChange === false) {
+      foundNonAcceleratableChange = true;
+    } else if (lambdaFunctionShortCircuitChange === true) {
+      // empty 'if' just for flow-aware typing to kick in...
+    } else {
+      acceleratableResources.push(lambdaFunctionShortCircuitChange);
+    }
+  });
+  return foundNonAcceleratableChange ? undefined : acceleratableResources;
+}
+
+function isAcceleratedLambdaFunctionChange(
+  logicalId: string, change: cfn_diff.ResourceDifference, assetParamsWithEnv: { [key: string]: string },
+): LambdaFunctionResource | boolean {
+  const lambdaCodeChange = isLambdaFunctionCodeOnlyChange(change, assetParamsWithEnv);
+  if (typeof lambdaCodeChange === 'boolean') {
+    return lambdaCodeChange;
+  } else {
+    // verify that the Asset changed - otherwise,
+    // it's a code-only change,
+    // but not to an asset
+    if (!assetMetadataChanged(change)) {
+      return false;
+    }
+
+    let functionPhysicalName: string | undefined;
+    try {
+      functionPhysicalName = stringifyPotentialCfnExpression(change.newValue?.Properties?.FunctionName, assetParamsWithEnv);
+    } catch (e) {
+      // It's possible we can't evaluate the function's name -
+      // for example, it can use a Ref to a different resource,
+      // which we wouldn't have in `assetParamsWithEnv`.
+      // That's fine though - ignore any errors,
+      // and treat this case the same way as if the name wasn't provided at all,
+      // which means it will be looked up using the listStackResources() call
+      // by the later phase (which actually does the Lambda function update)
+      functionPhysicalName = undefined;
+    }
+
+    return {
+      logicalId,
+      physicalName: functionPhysicalName,
+      code: lambdaCodeChange,
+    };
+  }
+}
+
+function isLambdaFunctionCodeOnlyChange(
+  change: cfn_diff.ResourceDifference, assetParamsWithEnv: { [key: string]: string },
+): LambdaFunctionCode | boolean {
+  if (!change.newValue) {
+    return false;
+  }
+  const newResourceType = change.newValue.Type;
+  // Ignore Metadata changes
+  if (newResourceType === 'AWS::CDK::Metadata') {
+    return true;
```

Comment:
?? return `true`?

Reply:
Explained in https://github.com/aws/aws-cdk/pull/15748#discussion_r677028402, I added an additional comment to the function.